specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,267 @@
+  - [Implementation](#implementation)
+    - [`constructor`](#constructor)
+    - [`initialize`](#initialize-1)
+    - [`anchors` / `getLatestAnchorState`](#anchors--getlatestanchorstate)
+    - [`registerMaybeValidGame`](#registermaybevalidgame-1)
+    - [`updateLatestValidGame`](#updatelatestvalidgame-1)
+    - [`tryUpdateLatestValidGame`](#tryupdatelatestvalidgame-1)
+    - [`setGameBlacklisted`](#setgameblacklisted-1)
+    - [`setRespectedGameType`](#setrespectedgametype-1)
+    - [`isGameInvalid`](#isgameinvalid-1)
+    - [`isGameValid`](#isgamevalid-1)
+    - [`disputeGameFinalityDelaySeconds`](#disputegamefinalitydelayseconds)
+    - [`disputeGameFactory`](#disputegamefactory-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+### Perspective
+
+Multiple contracts in the fault proof system have critical dependencies on things outside them:
+
+- The Portal needs to know whether a withdrawal's proof is based on a **valid** dispute game.
+- A new dispute game needs to initialize with the **latest valid anchor state**.
+- An existing dispute game needs to know whether it is **invalid**, so it can refund its bonds.
+
+The AnchorStateRegistry is these contracts' source of truth, managing and exposing dispute game and anchor state
+validity to moderate dispute games and withdrawals.
+
+Furthermore, the AnchorStateRegistry is a crucial player in incident response. It can invalidate dispute games, thereby
+invalidating withdrawals and dispute games founded on an incorrect root claim.
+
+## Definitions
+
+- **Anchor state**
+  - See [Fault Dispute Game -> Anchor State](fault-dispute-game.md#anchor-state).
+- **Authorized input**
+  - An input for which there is social consensus, i.e. coming from governance.
+- **Blacklisted game**
+  - A dispute game is blacklisted if it is set as blacklisted via **authorized input**.
+- **Validity timestamp**
+  - The validity timestamp is a timestamp internal to the contract that partly determines game validity and can only be
+    adjusted via **authorized input**.
+- **Invalid game**
+  - A dispute game is invalid if any of the following are true:
+    - Game was not created by the dispute game factory.
+    - Game was not created while it was the respected game type.
+    - Game is **blacklisted**.
+    - Game was created before the **validity timestamp**.
+    - Game status is `CHALLENGER_WINS`.
+- **Finalized game**
+  - A dispute game is finalized if all of the following are true:
+    - Game status is `CHALLENGER_WINS` or `DEFENDER_WINS`.
+    - Game `resolvedAt` timestamp is not zero.
+    - Game `resolvedAt` timestamp is more than `dispute game finality delay` seconds ago.
+- **Maybe valid game**
+  - A dispute game that is not an **invalid game** (but not yet a **finalized game**).
+- **Valid game**
+  - A game is a **Valid game** if it is not an **Invalid game**, and is a **Finalized game**.
+- **Latest valid game**
```

Comment:
Minor tweaks here:
- Can remove the bit about anchor state
- Want to somehow clearly define that it isn't the absolutely latest valid game but it's the latest valid game that the contract knows about